aws-sns-topicpolicy/src/test/java/software/amazon/sns/topicpolicy/CreateHandlerTest.java

```diff
@@ -157,55 +202,107 @@ public void handleRequest_Failure_Empty_policyDocument() {
         assertThrows(CfnInvalidRequestException.class, () -> handler.handleRequest(proxy, request, new CallbackContext(), proxyClient, logger));
     }
 
-
     @Test
-    public void handleRequest_Failure_Empty_Topics() {
+    public void handleRequest_Failure_InvalidParameterException() {
+
         final List<String> topics = new ArrayList<>();
+        topics.add("arn:aws:sns:us-east-1:123456789:my-topic110");
+        topics.add("arn:aws:sns:us-east-1:123456789:my-topic220");
+
         Map<String, Object> policyDocument = getSNSPolicy();
+
         final ResourceModel model = ResourceModel.builder()
                 .id("aws-sns-topic-policy-id-CfnInvalidRequestException")
                 .topics(topics)
                 .policyDocument(policyDocument)
                 .build();
+
         final ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel> builder()
                 .desiredResourceState(model)
                 .build();
 
-        assertThrows(CfnInvalidRequestException.class, () -> handler.handleRequest(proxy, request, null, proxyClient, logger));
+        when(proxyClient.client().setTopicAttributes(any(SetTopicAttributesRequest.class)))
+                        .thenThrow(InvalidParameterException.class);
+        final ProgressEvent<ResourceModel, CallbackContext> response = handler.handleRequest(proxy, request,
+                new CallbackContext(), proxyClient, logger);
+        assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
+        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.InternalFailure);
```

Comment:
Can you verify the times `SetTopicAttributes` is called in this case for verifying the backoff logic?

Reply:
What's the backoff logic? I can add a value to count it.